.github/workflows/ci.yaml

```diff
@@ -13,6 +14,13 @@ env:
   CARGO_PROFILE_TEST_DEBUG: 0
   CROSS_CONTAINER_UID: 0
 
+permissions:
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
```

Comment:
Is the head_ref the commit sha? If so, won't this be different every time? I'd have thought that maybe the PR number would be what we want

Reply:
Oh fab!!